ip.go

```diff
@@ -273,6 +289,25 @@ func (i *ProjectIPServiceOp) Request(projectID string, ipReservationReq *IPReser
 	return ipr, resp, err
 }
 
+// Update updates an existing IP reservation.
+func (i *ProjectIPServiceOp) Update(reservationID string, updateRequest *IPAddressUpdateRequest) (*IPAddressReservation, *Response, error) {
+	if validateErr := ValidateUUID(reservationID); validateErr != nil {
+		return nil, nil, validateErr
+	}
+	opts := &GetOptions{}
```

Comment:
We don't do this on any of our PATCH/PUT endpoints today, but we should allow for Options to be specified. This would allow, for example, ?include=... and ?exclude=... which would affect the resulting object. 

This would also enable use of arbitrary query string arguments that may be supported.

Reply:
Shouldn't that be an argument?

_api_call_options.go_
`type UpdateOptions = GetOptions`
...
`func (i *ProjectIPServiceOp) Update(reservationID string, updateRequest *IPAddressUpdateRequest, opts *UpdateOptions)`

Reply:
I'll add it, especially since you noticed a difference in behavior between offering CustomData in the body vs the query parameters.